### PR TITLE
don't duplicate cases when episode originating from eNikshay already exists

### PR DESCRIPTION
**Pull request by nickpell**
🏢 or 🐡 

@proteusvacuum cc: @esoergel

### CODE REVIEW

**Comment by nickpell** [comment]
✅

File: custom/enikshay/nikshay_datamigration/factory.py [hunk 1]
```diff
@@ -53,44 +55,44 @@ def nikshay_id(self):
...
         """
-        Get the existing person case for this nikshay ID, or None if no person case exists
+        Get the existing episode case for this nikshay ID, or None if no episode case exists
         """
-        matching_external_ids = self.case_accessor.get_cases_by_external_id(self.nikshay_id, case_type='person')
+        matching_external_ids = self.case_accessor.get_cases_by_external_id(
+            self.nikshay_id, case_type=EPISODE_CASE_TYPE
+        )
         if matching_external_ids:
             assert len(matching_external_ids) == 1
```
Comment (proteusvacuum) [inline custom/enikshay/nikshay_datamigration/factory.py]: I think there is the possibility that more than one is returned?

File: custom/enikshay/nikshay_datamigration/factory.py [hunk 2]
```diff
@@ -247,6 +248,7 @@ def get_episode_case_structure(self, occurrence_structure):
             'attrs': {
                 'case_type': EPISODE_CASE_TYPE,
                 'date_opened': self.patient_detail.pregdate1,
+                'external_id': self.nikshay_id,
```
Comment (proteusvacuum) [inline custom/enikshay/nikshay_datamigration/factory.py]: 👍

File: custom/enikshay/nikshay_datamigration/tests/test_create_enikshay_cases.py [hunk 3]
```diff
@@ -275,6 +276,45 @@ def test_outcome_died(self):
         drtb_hiv_referral_case_ids = self.case_accessor.get_case_ids_in_domain(type='drtb-hiv-referral')
         self.assertEqual(0, len(drtb_hiv_referral_case_ids))
 
+    def test_nikshay_case_from_enikshay_not_duplicated(self):
+        call_command('create_enikshay_cases', self.domain)
+        person_case_ids = self.case_accessor.get_case_ids_in_domain(type='person')
+        assert len(person_case_ids) == 1
```
Comment (proteusvacuum) [inline custom/enikshay/nikshay_datamigration/tests/test_create_enikshay_cases.py]: nit: `self.assertEqual` is used elsewhere in your tests

**Review by proteusvacuum — APPROVED** [review]

File: custom/enikshay/nikshay_datamigration/factory.py (hunk 1 above)
Comment (nickpell) [inline custom/enikshay/nikshay_datamigration/factory.py]: how could we get into that state?

**Pull request merged**